packages/streamline/src/components/inputs/input-text/input-text.tsx

```diff
@@ -0,0 +1,172 @@
+      handleBlur,
+      handleChangeText,
+      handleSecureEntry,
+    } = useTextInput(props);
+
+    React.useImperativeHandle(forwardedRef, () => ({
+      isFocused: () => {
+        return root ? root.isFocused() : false;
+      },
+      setNativeProps: (args: Record<string, unknown>) => {
+        return root && root.setNativeProps(args);
+      },
+      clear: () => {
+        return root && root.clear();
+      },
+
+      focus: () => {
+        return root && root.focus();
+      },
+
+      blur: () => {
+        return root && root.blur();
+      },
+    }));
+
+    const forceFocus = () => {
+      return root?.focus();
+    };
+    return (
+      <InputTextPrimitive
+        {...props}
+        forceFocus={forceFocus}
+        innerRef={(ref: any) => {
+          root = ref;
+        }}
+        accessibilityLabel={props.label}
+        isDisabled={isDisabled || isLoading}
+        isFocused={innerFocused}
+        onBlur={handleBlur}
+        onChangeText={handleChangeText}
+        onFocus={handleFocus}
+        left={undefined}
+        right={getRightComponent(
+          props.unit,
+          props.rightIconName,
+          props.type,
+          pointerEvents,
+          secureEntry,
+          handleSecureEntry,
+          isLoading
+        )}
+        secureTextEntry={secureEntry}
+        value={value}
+        parentState={{
+          focused: innerFocused,
+        }}
+      />
+    );
+  }
+);
+
+const getRightComponent = (
+  unit?: string,
+  rightIconName?: IconsName,
+  inputType?: string,
+  pointerEvents?: ViewProps['pointerEvents'],
+  secureEntry?: boolean,
+  handleSecureEntry?: () => void,
+  isLoading?: boolean
+) => {
+  let rightComponent;
+  if (rightIconName) {
+    rightComponent = <InputTextIcon name={rightIconName} />;
+  }
+
+  if (inputType === 'calendar') {
+    rightComponent = (
+      <InputTextIcon
+        name="Calendar"
+        forceTextInputFocus={true}
+        pointerEvents={pointerEvents}
+      />
+    );
+  }
+  if (inputType === 'password') {
+    rightComponent = (
+      <InputTextIcon
+        name={secureEntry ? 'EyeClosed' : 'EyeOpen'}
+        onPress={handleSecureEntry}
+        forceTextInputFocus={false}
+        pointerEvents={pointerEvents}
+      />
+    );
+  }
+  if (inputType === 'select') {
+    rightComponent = (
+      <InputTextIcon
+        name="ChevronDown"
+        forceTextInputFocus={false}
+        pointerEvents={pointerEvents}
+        loading={isLoading}
+      />
+    );
+  }
+  if (isLoading) {
+    rightComponent = (
+      <InputTextIcon
+        name="ChevronDown"
+        forceTextInputFocus={false}
+        pointerEvents={pointerEvents}
+        loading={isLoading}
+      />
+    );
+  }
+
+  if (unit) {
+    rightComponent = (
+      <Box paddingLeft="xs" paddingTop="xxs" pointerEvents="none">
+        <Text variant="body" color="GREY_700">
+          {unit}
+        </Text>
+      </Box>
+    );
+  }
+  return rightComponent;
```

Comment:
Same comment as others, I found the usage of the let as a potential vector of confusion